### ISSUE TITLE
Feature: create/mutate users

### ISSUE THREAD

**Issue by hendrikeng** [markhuot/craftql#172]
It would be awesome to be able to mutate/create users and their fields/customfields via craftQl.

Thanks for the awesome work 💯

**Comment by markhuot** [comment]
FYI, this is in progress.

**Comment by markhuot** [comment]
This is now on `dev-master`. If you're able to try it out you'll want to try something along these lines,

```graphql
mutation {
  upsertUser(id:1, firstName:"Mark", lastName:"Huot") {
    id
    name # returns `Mark Huot` after the mutation
  }
}
```

You should be able to inspect the schema via GraphiQL if you're lost. You can search by user ID and edit any default fields as well as any custom fields applied to users. The return of the field is the edited user.

Note: there are new token scopes for this. If you have existing tokens you'll need to re-save the token with the new scopes enabled to make use of the new user mutation.

**Comment by j2is** [comment]
Question: Is it possible to mutate/create matrix blocks and elements in general?

**Comment by markhuot** [comment]
@j2is, can you open a new issue if you still have questions on mutating matrix blocks. It is possible and you should be able to infer the schema from GraphiQL. If you have questions though I'm happy to walk through some sample queries in a new ticket.

**Comment by markhuot** [comment]
Closing this issue. The fix is included in version 1.2.1.

**Comment by tristangemus** [comment]
Is there password field support yet?
